flake/flake.nix

```diff
@@ -0,0 +1,224 @@
+{
```

Comment:
Maybe put this files into `.garn/legacy-flake.nix` if possible, otherwise `.garn/legacy/flake.nix`?

Reply:
Maybe we should have two dirs, data and gen, to distinguish things we use from things we generate

Reply:
Maybe we should have two dirs, data and gen, to distinguish things we use from things we generate

Reply:
The dogfooding has been moved to #469